File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/c_solutions/2124-check-if-all-as-appears-before-all-bs.c
bool checkString(char* s) {
    int foundB = 0;
    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == 'b') {
            foundB = 1;
        } else if (s[i] == 'a' && foundB) {
            return false;
        }
    }
    return true;
}

File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/2124-check-if-all-as-appears-before-all-bs.js
var checkString = function(s) {
    let foundB = false;
    for (let char of s) {
        if (char === 'b') {
            foundB = true;
        } else if (foundB && char === 'a') {
            return false;
        }
    }
    return true;
};